ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/LedgerClient.scala

```diff
@@ -100,10 +103,28 @@ object LedgerClient {
     for {
       ledgerId <- new LedgerIdentityClient(
         stub(LedgerIdentityServiceGrpc.stub(channel), config.token)
-      )
-        .satisfies(config.ledgerIdRequirement)
+      ).satisfies(config.ledgerIdRequirement)
```

Comment:
sorry reshuffled my comment to the right place I misread the code first. I think it still applies though.

Reply:
> Which so far my understanding is should also work without token

No, `satisfies` does eagerly call the `LedgerIdentityService`. What changes is whether or not the user-provided ledger identifier will be compared with the actual one. If the Ledger API requires calls to be authorized, the `LedgerIdentityService` requires it as well.

> we just pick up the ledger id the user specified

This would require to know the ledger identifier beforehand, which we cannot rely upon.